### PR TITLE
Implemented the ButtonDropdown component

### PR DESCRIPTION
**Pull request by Wildhoney**
Reference: http://twitter.github.io/bootstrap/components.html#buttonDropdowns
- Updates when new items are added/old items are removed;
- Changes the "disabled" class name when the `disabled` property is toggled;
- Invokes an action on the controller as specified by the `actionName` property;
- Allows the specifying of custom classes, such as `btn-danger`;
- `EXTEND_PROTOTYPES` compliant;
- Documented in the README.md for developers;

### CODE REVIEW

**Comment by Wildhoney** [comment]
I'm not too sure why the Travis build fails. Any ideas?

**Comment by dmathieu** [comment]
The tests do not pass because your code does not pass the automated [jshint](http://www.jshint.com/) test.
At line 92, remove the `javascript:void(0);`, and replace it with a `#`.

**Comment by Wildhoney** [comment]
Thanks, Damien! Travis build now passing. I should have seen the reason for it failing, but when I was committing the code last night, I was awfully tired.

File: packages/ember-bootstrap/lib/views/button_dropdown.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     * @property classNames
+     * @type {Array}
+     */
+    classNames: ['btn-group'],
+
+    /**
+     * @property layout
+     * @type {Function}
+     */
+    layout: Ember.Handlebars.compile('<a {{bindAttr class="view.typeClass :btn :dropdown-toggle" }} data-toggle="dropdown" href="javascript:void(0);">{{view.label}}<span class="caret"></span></a>' +
```
Comment (dmathieu) [inline packages/ember-bootstrap/lib/views/button_dropdown.js]: Could you avoid having those string concatenation ?
Take a look at [how we do it in modal_pane](https://github.com/emberjs-addons/ember-bootstrap/blob/master/packages/ember-bootstrap/lib/views/modal_pane.js#L4).

**Comment by dmathieu** [comment]
Could you add a test which makes sure that adding/removing an element after adding the ButtonDropdown to the DOM works ?

**Comment by Wildhoney** [comment]
DOM tests added, and I've also used `defaultTemplate` instead of `template`/`layout` to keep it consistent with the other bootstrap elements.

**Comment by dmathieu** [comment]
That's awesome, thank you! :sunflower:
I'll merge this as soon as the travis build is passing.

**Comment by Wildhoney** [comment]
:relaxed: Excellent! Thanks.

**Pull request merged**